### ISSUE TITLE
Feature request: Live reload

### ISSUE THREAD

**Issue by keybits** [svenstaro/miniserve#1459]
Would you consider adding optional live reloading of files when they change?

I currently use [devd](https://github.com/cortesi/devd) for this. It's similar to miniserve in that it's a single binary for serving files over http. It has a live reload option which is very handy for local web development.

devd works fine, but development has stopped and miniserve has some features not available in devd. It'd be great to use just one tool.

**Comment by svenstaro** [comment]
I suppose you don't only want to live reload the file tree but also the contents of file. I'm not really sure how that would work. We can't live reload everything and we'd need to inject something into pages to make it work?

**Comment by keybits** [comment]
> we'd need to inject something into pages to make it work?

Yes, that's [how devd works](https://github.com/cortesi/devd?tab=readme-ov-file#livereload)

Since it's optional, it only happens when live reloading is enabled with sometthing like `-l` in the command invocation.

For filesystem watching devd uses it's own Go filesystem watching library [moddwatch](https://github.com/cortesi/moddwatch).
In the Rust world, I believe [notify](https://github.com/notify-rs/notify) would be a good option as it's battle-tested and cross plaform.

No problem if this is out of scope and would bring in too many dependencies.

**Comment by svenstaro** [comment]
I'm not opposed to this if it doesn't introduce too much new code. If you like, you can throw together a quick PoC and I can take a look at that with no promises made as to its inclusion.
